tests/test_create_peer_did_numalgo_2.py

```diff
@@ -0,0 +1,578 @@
+
+
+def test_create_numalgo_2_encryption_key_as_signing():
+    encryption_keys = [PublicKeyAgreement(encoded_value="6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH",
+                                          type=PublicKeyTypeAgreement.X25519, encoding_type=EncodingType.BASE58)]
+    services = '''{
+        "type": "didcommmessaging",
+        "serviceEndpoint": "https://example.com/endpoint",
+        "routingKeys": ["did:example:somemediator#somekey"]
+        }
+        '''
+    with pytest.raises(ValueError):
+        peer_did_algo_2 = create_peer_did_numalgo_2(encryption_keys=encryption_keys, signing_keys=encryption_keys,
+                                                    services=services)
+
+
+def test_create_numalgo_2_signing_key_as_encryption():
+    signing_keys = [
+        PublicKeyAuthentication(
+            encoded_value="zXwpBnMdCm1cLmKuzgESn29nqnonp1ioqrQMRHNsmjMypp",
+            type=PublicKeyTypeAuthentication.ED25519, encoding_type=EncodingType.BASE58),
+        PublicKeyAuthentication(
+            encoded_value="zx8xB2pv7cw8q1PdDacSrdWE3dtB9f7Nxk886mdzNFoPtY",
+            type=PublicKeyTypeAuthentication.ED25519, encoding_type=EncodingType.BASE58)]
+    services = '''{
+        "type": "didcommmessaging",
+        "serviceEndpoint": "https://example.com/endpoint",
+        "routingKeys": ["did:example:somemediator#somekey"]
+        }
+        '''
+    with pytest.raises(TypeError):
+        peer_did_algo_2 = create_peer_did_numalgo_2(encryption_keys=signing_keys, signing_keys=signing_keys,
+                                                    services=services)
+
+
+def test_create_numalgo_2_service_is_an_array():
+    encryption_keys = [PublicKeyAgreement(encoded_value="JhNWeSVLMYccCk7iopQW4guaSJTojqpMEELgSLhKwRr",
+                                          type=PublicKeyTypeAgreement.X25519, encoding_type=EncodingType.BASE58)]
+    signing_keys = [
+        PublicKeyAuthentication(
+            encoded_value="ByHnpUCFb1vAfh9CFZ8ZkmUZguURW8nSw889hy6rD8L7",
+            type=PublicKeyTypeAuthentication.ED25519, encoding_type=EncodingType.BASE58),
+        PublicKeyAuthentication(
+            encoded_value="3M5RCDjPTWPkKSN3sxUmmMqHbmRPegYP1tjcKyrDbt9J",
+            type=PublicKeyTypeAuthentication.ED25519, encoding_type=EncodingType.BASE58)]
+    services = '''
+            [
+                {
+                    "type": "didcommmessaging",
+                    "serviceEndpoint": "https://example.com/endpoint",
+                    "routingKeys": ["did:example:somemediator#somekey"]
+                },
+                {
+                    "type": "didcommmessaging",
+                    "serviceEndpoint": "https://example.com/endpoint2",
+                    "routingKeys": ["did:example:somemediator#somekey2"]
+                }
+            ]
+            '''
+
+    peer_did_algo_2 = create_peer_did_numalgo_2(encryption_keys=encryption_keys, signing_keys=signing_keys,
```

Comment:
It would be great to check expected peer_did value in that case to make sure that we appended only one .S for the whole service JSON, not two.